lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -15,16 +16,74 @@ class FirebaseAppDistributionAction < Action
       FIREBASECMD_ACTION = "appdistribution:distribute".freeze
       BASE_URL = "https://firebaseappdistribution.googleapis.com"
       PATH = "/v1alpha/apps/"
+      MAX_POLLING_RETRIES = 60
+      POLLING_INTERVAL_S = 2
 
       extend Helper::FirebaseAppDistributionHelper
 
       def self.run(params)
         params.values # to validate all inputs before looking for the ipa/apk
-        validate_app!(params[:app])
+        platform = Actions.lane_context[Actions::SharedValues::PLATFORM_NAME]
+        binary_path = params[:ipa_path] || params[:apk_path]
+
+        if params[:app] # Set app_id if it is specified as a parameter
+          app_id = params[:app]
+        else
+          if platform == :ios
+            archive_path = Actions.lane_context[SharedValues::XCODEBUILD_ARCHIVE]
+            if archive_patharchive_path
+              app_id = findout_ios_app_id_from_archive(archive_path)
+            end
+          end
+          if app_id.nil?
+            app_id = UI.input("Your app's Firebase App ID. You can find the App ID in the Firebase console, on the General Settings page: ")
```

Comment:
Just to call it out: this actually doesn't match the previous behavior, which seems to be that it would call out to the Firebase CLI without setting `--app`. At that point the Firebase CLI would error out: https://github.com/firebase/firebase-tools/blob/master/src/commands/appdistribution-distribute.ts#L18. Am I right?

I'm OK with prompting the user for it here, I think we should just recognize that it's a change and we should probably run it by Alex.

Reply:
Sounds good, I reverted it to the previous behavior for now